### PR TITLE
Remaning player by refernce clean up

### PR DESCRIPTION
**Pull request by AJenbo**
These are the last of the more trivial player by reference clean ups :)

### CODE REVIEW

File: Source/monster.cpp [hunk 1]
```diff
@@ -5083,7 +5083,7 @@ void TalktoMonster(MonsterStruct &monster)
 
 void SpawnGolum(int i, Point position, int mi)
 {
-	assert(i >= 0 && i < MAXMONSTERS);
+	assert(i >= 0 && i < MAX_PLRS);
```
[the remark this answers is not in the document]
Comment (AJenbo) [reply Source/monster.cpp:5086]: Yeah the first `MAX_PLRS` monsters are reserved for the players golem., I suppose we can check that `MAX_PLRS <= MAXMONSTERS` but that seams a bit redundant :)

[the remark this answers is not in the document]
Comment (qndel) [reply Source/monster.cpp:5086]: ye but if it's < MAX_PLRS, then it's < MAXMONSTERS too ;p

[the remark this answers is not in the document]
Comment (obligaron) [reply Source/monster.cpp:5086]: Good so no one is working on the same mod as I do ... my MMO-PRPG Diablo 1 :zany_face:

[the remark this answers is not in the document]
Comment (AJenbo) [reply Source/monster.cpp:5086]: :joy:

**Review by obligaron — APPROVED** [review]

**Pull request merged**